osquery/tables/cloud/azure/azure_instance_metadata.cpp

```diff
@@ -0,0 +1,53 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/utils/azure/azure_util.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genAzureMetadata(QueryContext& context) {
+  QueryData results;
+  Row r;
+
+  JSON doc;
+
+  Status s = fetchAzureMetadata(doc);
+
+  if (!s.ok()) {
```

Comment:
Also check that `doc.doc().isObject()`

Reply:
Thanks! I'll do that in `fetchAzureMetadata`, to avoid duplication.